### PR TITLE
En 5851 fix golangci linter issues

### PR DESCRIPTION
**Pull request by miiu96**

### CODE REVIEW

File: integrationTests/mock/vmContainerMock.go [hunk 1]
```diff
@@ -71,8 +71,5 @@ func (V *VMContainerMock) Keys() [][]byte {
 
 // IsInterfaceNil returns true if there is no value under the interface
 func (V *VMContainerMock) IsInterfaceNil() bool {
-	if V == nil {
-		return true
-	}
-	return false
+	return V == nil
```
Comment (AdoAdoAdo) [inline integrationTests/mock/vmContainerMock.go]: please rename the receiver to vmc in all methods for this mock

File: p2p/libp2p/directSender_test.go [hunk 2]
```diff
@@ -69,7 +69,10 @@ func createLibP2PCredentialsDirectSender() (peer.ID, libp2pCrypto.PrivKey) {
 func TestNewDirectSender_NilContextShouldErr(t *testing.T) {
 	hs := &mock.ConnectableHostStub{}
 
-	ds, err := libp2p.NewDirectSender(nil, hs, func(msg p2p.MessageP2P) error {
+	ctx := context.Background()
+	_ = ctx
```
Comment (BeniaminDrasovean) [inline p2p/libp2p/directSender_test.go]: Why is this needed here?

File: p2p/libp2p/libp2pContext_test.go [hunk 3]
```diff
@@ -11,7 +11,10 @@ import (
 )
 
 func TestNewLibp2pContext_NilContextShoulsErr(t *testing.T) {
-	lctx, err := libp2p.NewLibp2pContext(nil, &mock.ConnectableHostStub{})
+	ctx := context.Background()
+	_ = ctx
```
Comment (BeniaminDrasovean) [inline p2p/libp2p/libp2pContext_test.go]: Why is this needed here?

File: p2p/libp2p/netMessenger_test.go [hunk 4]
```diff
@@ -132,7 +132,10 @@ func containsPeerID(list []p2p.PeerID, searchFor p2p.PeerID) bool {
 func TestNewMemoryLibp2pMessenger_NilContextShouldErr(t *testing.T) {
 	netw := mocknet.New(context.Background())
 
-	mes, err := libp2p.NewMemoryMessenger(nil, netw, discovery.NewNullDiscoverer())
+	ctx := context.Background()
+	_ = ctx
```
Comment (BeniaminDrasovean) [inline p2p/libp2p/netMessenger_test.go]: Why is this needed here?

File: p2p/libp2p/netMessenger_test.go [hunk 5]
```diff
@@ -191,8 +194,11 @@ func TestNewNetworkMessenger_NilContextShouldErr(t *testing.T) {
 
 	_, sk := createLibP2PCredentialsMessenger()
 
+	ctx := context.Background()
+	_ = ctx
```
Comment (BeniaminDrasovean) [inline p2p/libp2p/netMessenger_test.go]: Why is this needed here?

File: process/dataValidators/nilHeaderValidator_test.go [hunk 6]
```diff
@@ -24,14 +24,3 @@ func TestNilHeaderValidator_IsHeaderValidForProcessing(t *testing.T) {
 
 	assert.Nil(t, nhv.HeaderValidForProcessing(nil))
 }
-
-//------- IsInterfaceNil
-
-func TestNilHeaderValidator_IsInterfaceNil(t *testing.T) {
```
Comment (BeniaminDrasovean) [inline process/dataValidators/nilHeaderValidator_test.go]: Why have you deleted this test?

File: process/dataValidators/txValidator_test.go [hunk 7]
```diff
@@ -252,16 +251,3 @@ func TestTxValidator_CheckTxValidityTxIsOkShouldReturnTrue(t *testing.T) {
 	result := txValidator.CheckTxValidity(txValidatorHandler)
 	assert.Nil(t, result)
 }
-
-//------- IsInterfaceNil
-
-func TestTxValidator_IsInterfaceNil(t *testing.T) {
```
Comment (BeniaminDrasovean) [inline process/dataValidators/txValidator_test.go]: Why have you deleted this test?

File: process/sync/metablock_test.go [hunk 8]
```diff
@@ -1177,6 +1177,7 @@ func TestMetaBootstrap_RollBackIsEmptyCallRollBackOneBlockOkValsShouldWork(t *te
 				//bytes represent a tx block body (strings are returns from txBlockUnit.Get which is also a stub here)
 				//copy only defined fields
 				obj = prevTxBlockBody
+				_ = obj
```
Comment (BeniaminDrasovean) [inline process/sync/metablock_test.go]: Why is this needed here?

File: process/sync/metablock_test.go [hunk 9]
```diff
@@ -1311,6 +1312,7 @@ func TestMetaBootstrap_RollBackIsEmptyCallRollBackOneBlockToGenesisShouldWork(t
...
```
Comment (BeniaminDrasovean) [inline process/sync/metablock_test.go]: Why is this needed here?

File: process/sync/shardblock_test.go [hunk 10]
```diff
@@ -1405,6 +1405,7 @@ func TestBootstrap_RollBackIsEmptyCallRollBackOneBlockOkValsShouldWork(t *testin
 				//bytes represent a tx block body (strings are returns from txBlockUnit.Get which is also a stub here)
 				//copy only defined fields
 				obj = prevTxBlockBody
+				_ = obj
```
Comment (BeniaminDrasovean) [inline process/sync/shardblock_test.go]: Why is this needed here?

File: process/sync/shardblock_test.go [hunk 11]
```diff
@@ -1550,6 +1551,7 @@ func TestBootstrap_RollbackIsEmptyCallRollBackOneBlockToGenesisShouldWork(t *tes
...
```
Comment (BeniaminDrasovean) [inline process/sync/shardblock_test.go]: Why is this needed here?

File: p2p/libp2p/directSender_test.go [hunk 12]
```diff
@@ -69,7 +69,10 @@ func createLibP2PCredentialsDirectSender() (peer.ID, libp2pCrypto.PrivKey) {
 func TestNewDirectSender_NilContextShouldErr(t *testing.T) {
 	hs := &mock.ConnectableHostStub{}
 
-	ds, err := libp2p.NewDirectSender(nil, hs, func(msg p2p.MessageP2P) error {
+	ctx := context.Background()
```
Comment (AdoAdoAdo) [inline p2p/libp2p/directSender_test.go]: you can replace these 3 lines with a simple declaration

File: p2p/libp2p/libp2pContext_test.go [hunk 13]
```diff
@@ -11,7 +11,10 @@ import (
 )
 
 func TestNewLibp2pContext_NilContextShoulsErr(t *testing.T) {
-	lctx, err := libp2p.NewLibp2pContext(nil, &mock.ConnectableHostStub{})
+	ctx := context.Background()
```
Comment (AdoAdoAdo) [inline p2p/libp2p/libp2pContext_test.go]: please replace the three lines with declaration

File: p2p/libp2p/netMessenger_test.go [hunk 14]
```diff
@@ -132,7 +132,10 @@ func containsPeerID(list []p2p.PeerID, searchFor p2p.PeerID) bool {
 func TestNewMemoryLibp2pMessenger_NilContextShouldErr(t *testing.T) {
 	netw := mocknet.New(context.Background())
 
-	mes, err := libp2p.NewMemoryMessenger(nil, netw, discovery.NewNullDiscoverer())
+	ctx := context.Background()
```
Comment (AdoAdoAdo) [inline p2p/libp2p/netMessenger_test.go]: please replace the three lines with variable declaration

File: p2p/libp2p/netMessenger_test.go [hunk 15]
```diff
@@ -191,8 +194,11 @@ func TestNewNetworkMessenger_NilContextShouldErr(t *testing.T) {
 
 	_, sk := createLibP2PCredentialsMessenger()
 
+	ctx := context.Background()
```
Comment (AdoAdoAdo) [inline p2p/libp2p/netMessenger_test.go]: replace with variable declaration

File: process/dataValidators/nilHeaderValidator_test.go [hunk 16]
```diff
@@ -24,14 +24,3 @@ func TestNilHeaderValidator_IsHeaderValidForProcessing(t *testing.T) {
...
-	t.Parallel()
-
-	hdrValidator, _ := dataValidators.NewNilHeaderValidator()
```
Comment (AdoAdoAdo) [inline process/dataValidators/nilHeaderValidator_test.go]: please replace with declaration and leave the assert

File: process/dataValidators/txValidator_test.go [hunk 17]
```diff
@@ -252,16 +251,3 @@ func TestTxValidator_CheckTxValidityTxIsOkShouldReturnTrue(t *testing.T) {
...
-	t.Parallel()
-
-	accounts := getAccAdapter(0, big.NewInt(0))
-	shardCoordinator := createMockCoordinator("_", 0)
-	txValidator, _ := dataValidators.NewTxValidator(accounts, shardCoordinator, 100)
-	txValidator = nil
```
Comment (AdoAdoAdo) [inline process/dataValidators/txValidator_test.go]: please replace with declaration and leave the assert

File: process/mock/txProcessorMock.go [hunk 18]
```diff
@@ -31,8 +31,5 @@ func (etm *TxProcessorMock) ProcessSmartContractResult(scr *smartContractResult.
 
 // IsInterfaceNil returns true if there is no value under the interface
 func (etm *TxProcessorMock) IsInterfaceNil() bool {
```
Comment (AdoAdoAdo) [inline process/mock/txProcessorMock.go]: please rename the receiver for all methods in this mock

File: process/sync/metablock_test.go (hunk 8 above)
Comment (AdoAdoAdo) [inline process/sync/metablock_test.go]: this is masking an error.
please do a copy of the prevTxBlockBody fields into obj

File: process/sync/metablock_test.go (hunk 9 above)
Comment (AdoAdoAdo) [inline process/sync/metablock_test.go]: same as above comment

File: process/sync/shardblock_test.go (hunk 10 above)
Comment (AdoAdoAdo) [inline process/sync/shardblock_test.go]: this is masking an error
please do a deep copy

File: process/sync/shardblock_test.go (hunk 11 above)
Comment (AdoAdoAdo) [inline process/sync/shardblock_test.go]: this is masking an error
please do a deep copy

File: data/block/block_test.go [hunk 19]
```diff
@@ -245,6 +245,7 @@ func TestBody_IntegrityAndValidityNil(t *testing.T) {
 	t.Parallel()
 
 	body := block.Body{}
+	_ = body
```
[the remark this answers is not in the document]
Comment (miiu96) [reply data/block/block_test.go]: done.

File: p2p/libp2p/directSender_test.go (hunk 2 above)
Comment (miiu96) [reply p2p/libp2p/directSender_test.go]: done.

File: p2p/libp2p/libp2pContext_test.go (hunk 3 above)
Comment (miiu96) [reply p2p/libp2p/libp2pContext_test.go]: done.

File: p2p/libp2p/netMessenger_test.go (hunk 4 above)
Comment (miiu96) [reply p2p/libp2p/netMessenger_test.go]: done.

File: p2p/libp2p/netMessenger_test.go (hunk 5 above)
Comment (miiu96) [reply p2p/libp2p/netMessenger_test.go]: done.

File: process/dataValidators/nilHeaderValidator_test.go (hunk 6 above)
Comment (miiu96) [reply process/dataValidators/nilHeaderValidator_test.go]: test was restored.

File: process/dataValidators/txValidator_test.go (hunk 7 above)
Comment (miiu96) [reply process/dataValidators/txValidator_test.go]: test was restored.

File: consensus/mock/marshalizerMock.go [hunk 20]
```diff
@@ -54,8 +54,5 @@ func (MarshalizerMock) Version() string {
 
 // IsInterfaceNil returns true if there is no value under the interface
 func (mm MarshalizerMock) IsInterfaceNil() bool {
-	if &mm == nil {
-		return true
-	}
-	return false
+	return &mm == nil
```
[the remark this answers is not in the document]
Comment (miiu96) [reply consensus/mock/marshalizerMock.go]: done.

File: consensus/mock/shardCoordinatorMock.go [hunk 21]
```diff
@@ -45,8 +45,5 @@ func (scm ShardCoordinatorMock) CommunicationIdentifier(destShardID uint32) stri
 
 // IsInterfaceNil returns true if there is no value under the interface
 func (scm ShardCoordinatorMock) IsInterfaceNil() bool {
-	if &scm == nil {
-		return true
-	}
-	return false
+	return &scm == nil
```
[the remark this answers is not in the document]
Comment (miiu96) [reply consensus/mock/shardCoordinatorMock.go]: done.

File: core/mock/shardCoordinatorMock.go [hunk 22]
```diff
@@ -45,8 +45,5 @@ func (scm ShardCoordinatorMock) CommunicationIdentifier(destShardID uint32) stri
 
 // IsInterfaceNil returns true if there is no value under the interface
 func (scm ShardCoordinatorMock) IsInterfaceNil() bool {
-	if &scm == nil {
-		return true
-	}
-	return false
+	return &scm == nil
```
[the remark this answers is not in the document]
Comment (miiu96) [reply core/mock/shardCoordinatorMock.go]: done.

File: data/block/block_test.go (hunk 19 above)
[the remark this answers is not in the document]
Comment (miiu96) [reply data/block/block_test.go]: done.

File: integrationTests/mock/vmContainerMock.go (hunk 1 above)
Comment (miiu96) [reply integrationTests/mock/vmContainerMock.go]: done.

File: p2p/libp2p/directSender_test.go (hunk 12 above)
Comment (miiu96) [reply p2p/libp2p/directSender_test.go]: done.

File: p2p/libp2p/libp2pContext_test.go (hunk 13 above)
Comment (miiu96) [reply p2p/libp2p/libp2pContext_test.go]: done.

File: p2p/libp2p/netMessenger_test.go (hunk 14 above)
Comment (miiu96) [reply p2p/libp2p/netMessenger_test.go]: done.

File: p2p/libp2p/netMessenger_test.go (hunk 15 above)
Comment (miiu96) [reply p2p/libp2p/netMessenger_test.go]: done.

File: process/dataValidators/nilHeaderValidator_test.go (hunk 16 above)
Comment (miiu96) [reply process/dataValidators/nilHeaderValidator_test.go]: here i cannot replace with a declaration because NewNilHeaderValidator will return a private type ... so to fix the warning i will write this line of code " _ = hdrValidator"

File: process/dataValidators/txValidator_test.go (hunk 17 above)
Comment (miiu96) [reply process/dataValidators/txValidator_test.go]: here i cannot replace with a declaration because NewTxValidator will return a private type ... so to fix the warning i will write this line of code " _ = txValidator"

File: process/mock/txProcessorMock.go (hunk 18 above)
Comment (miiu96) [reply process/mock/txProcessorMock.go]: done.

File: process/sync/metablock_test.go (hunk 8 above)
Comment (miiu96) [reply process/sync/metablock_test.go]: done.

File: process/sync/metablock_test.go (hunk 9 above)
Comment (miiu96) [reply process/sync/metablock_test.go]: done.

File: process/sync/shardblock_test.go (hunk 10 above)
Comment (miiu96) [reply process/sync/shardblock_test.go]: done.

File: process/sync/shardblock_test.go (hunk 11 above)
Comment (miiu96) [reply process/sync/shardblock_test.go]: done.

Comment (miiu96) [reply process/sync/shardblock_test.go]: refactored.

File: process/sync/shardblock_test.go (hunk 10 above)
Comment (miiu96) [reply process/sync/shardblock_test.go]: refactored.
